Source/OCMock/NSInvocation+OCMAdditions.m

```diff
@@ -322,18 +324,25 @@ - (id)getArgumentAtIndexAsObject:(NSInteger)argIndex
                     return copiedStructs[argIndex];
                 }
             }
-            return [self getStructAtIndexAsNSValue:argIndex];
+            id _Nullable nsValue = [self getStructAtIndexAsNSValue:argIndex];
+            if (nsValue) {
+                return nsValue;
+            }
         }
     }
     [NSException raise:NSInvalidArgumentException format:@"Argument type '%s' not supported", argType];
     return nil;
 }
 
 
-- (NSValue *)getStructAtIndexAsNSValue:(NSInteger)argIndex {
+- (nullable NSValue *)getStructAtIndexAsNSValue:(NSInteger)argIndex {
     NSUInteger argSize;
     const char *argType = OCMTypeWithoutQualifiers([[self methodSignature] getArgumentTypeAtIndex:(NSUInteger)argIndex]);
-    NSGetSizeAndAlignment([[self methodSignature] getArgumentTypeAtIndex:(NSUInteger)argIndex], &argSize, NULL);
+    @try {
+        NSGetSizeAndAlignment([[self methodSignature] getArgumentTypeAtIndex:(NSUInteger)argIndex], &argSize, NULL);
```

Comment:
@ashdnazg 
1. The only thing which trows here is `getArgumentTypeAtIndex:`. I don't see how this can happen.
2. It just exchanges one exception raised with another in `getArgumentAtIndexAsObject`.
3. Which tests failed in foundation?

Reply:
1. I don't think you are correct, it's `NSGetSizeAndAlignment` that throws.
2. Not when it's called through the `retain` function.
3. I don't remember at the moment which one, something that used `lt::Quad`.

Reply:
The problematic struct is `{Quad={array<CGPoint, 4UL>=[4{CGPoint=dd}]}{optional<lt::Quad::Type>}}`